pkg/kubectl/cmd/replace.go

```diff
@@ -200,17 +210,37 @@ func forceReplace(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 	if err != nil {
 		return err
 	}
+
 	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
 		ContinueOnError().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, options).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
+		Latest().
```

Comment:
@liggitt The [first `result` in line 214](https://github.com/kubernetes/kubernetes/pull/41924/files/6ccbdbb3797be28746660e76b729cba9b5d4ac98?diff=split#diff-1569293db00490c928c40f53fc14800cR214) is built to fetch and save the last-applied-config annotation. The [second `result` in line 232](https://github.com/kubernetes/kubernetes/pull/41924/files/6ccbdbb3797be28746660e76b729cba9b5d4ac98?diff=split#diff-1569293db00490c928c40f53fc14800cR232) is used for deleting the resources. It should be OK if the resource changes but the last-applied-config annotation is not changed. Is this OK for you?
Or do you have a better way to do this?